src/eventHandlers/FunctionEnvironmentReloadHandler.ts

```diff
@@ -34,7 +34,11 @@ export class FunctionEnvironmentReloadHandler extends EventHandler<
             logCategory: LogCategory.System,
         });
 
-        process.env = Object.assign({}, msg.environmentVariables);
+        // reset existing env vars
+        Object.keys(process.env).map((key) => delete process.env[key]);

```

Comment:
We "reset" existing env vars before your change, right? Since we completely overwrote process.env with a new object? By itself, this line feels dangerous - but assuming we're matching previous behavior I think it makes sense

Reply:
Yes, and the current unit tests ensure that behavior, e.g. https://github.com/Azure/azure-functions-nodejs-worker/blob/ad99f5d4f5613e42c7630b39d9f24a849116d9c6/test/eventHandlers/FunctionEnvironmentReloadHandler.test.ts#L111